tests/test_static_slots.rs

```diff
@@ -0,0 +1,57 @@
+#![cfg(feature = "macros")]
+
+use pyo3::exceptions::PyIndexError;
+use pyo3::prelude::*;
+use pyo3::types::IntoPyDict;
+
+use pyo3::py_run;
+
+mod common;
+
+#[pyclass]
+struct Vector3 {
+    elements: [f64; 3],
+}
+
+#[pymethods]
+impl Vector3 {
+    #[new]
+    fn new(x: f64, y: f64, z: f64) -> Self {
+        Self {
+            elements: [x, y, z],
+        }
+    }
+
+    #[staticmethod]
+    fn __len__() -> usize {
+        3
+    }
+
+    fn __getitem__(&self, idx: isize) -> PyResult<f64> {
```

Comment:
Why not test a "selfless" `__getitem__` as well?

Reply:
yep, good suggestion, pushed tests for a very useful `Count5` class